src/API/Reports/Customers/DataStore.php

```diff
@@ -126,6 +126,46 @@ public static function sync_on_order_delete( $order_id, $customer_id ) {
 		}
 	}
 
+	/**
+	 * Sync customers data after an order was updated.
+	 *
+	 * Only updates customer if it is the customers last order.
+	 *
+	 * @param int $post_id of order.
+	 */
+	public static function sync_order_customer( $post_id ) {
+		global $wpdb;
+
+		if ( 'shop_order' !== get_post_type( $post_id ) && 'shop_order_refund' !== get_post_type( $post_id ) ) {
+			return -1;
+		}
+
+		$order       = wc_get_order( $post_id );
+		$customer_id = self::get_existing_customer_id_from_order( $order );
+		if ( ! $customer_id ) {
+			return -1;
+		}
+		$customer   = new \WC_Customer( $customer_id );
```

Comment:
I know this is a bit confusing, but `WC_Customer` expects a WordPress user ID, not WooCommerce Admin's generated customer ID. I have enough test data that there was actually a collision on IDs which caused the customer data I was updating to not propagate to the customers table.

_P.S. Sorry for the after-the-fact review here!_